Add empty stubs for xml.dom.* modules.

Flameeyes opened the pull request:
While this does not add any interface in the modules, it tells mypy that
the modules _exist_, which is a starting point.

srittau commented:
Not merging for now, since it conflicts with #3852.

Also, empty files should instead have a top-level `__getattr__` definition per our CONTRIBUTING file:

```python
def __getattr__(name: str) -> Any: ...  # incomplete
```

Flameeyes commented:
Hah! Thanks for the pointer, I did miss that.

And funny how timing worked out with this. I'll retract this pull request once I'm in front of the computer.

On Tue, Mar 17, 2020, at 13:36, Sebastian Rittau wrote:
> 

> Not merging for now, since it conflicts with #3852 <https://github.com/python/typeshed/pull/3852>.

> Also, empty files should instead have a top-level `__getattr__` definition per our CONTRIBUTING file:

> def __getattr__(name: str) -> Any: ...  # incomplete

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/python/typeshed/pull/3856#issuecomment-600075105>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAASIUR3EA6B42BMMJ42JX3RH54FVANCNFSM4LNH47VQ>.

srittau commented:
The PR is fine for minidom.pyi and pulldom.pyi, just __init__.pyi conflicts.

Flameeyes commented:
Removed the `__init__.pyi` file and updated the empty stubs.

It's not clear to me if the `__init__` is needed for this, but worse case I guess I'll wait for #3852 and then fast-forward this request.

JelleZijlstra commented:
You still need to add the `__getattr__` to the empty files.

Flameeyes commented:
Yeah I was waiting to hear on whether I should need for the other one. Added now though.

Flameeyes commented:
Is it good to go now?

srittau commented:
Looks good, still waiting for #3852, though, since that adds the `__init__.pyi` file. @jermenkoo

jermenkoo commented:
I'll update the PR today!
